test/e2e_node/pod_resize_test.go

```diff
@@ -307,56 +309,164 @@ func verifyPodContainersCgroupValues(ctx context.Context, f *framework.Framework
+		if err != nil {
+			return err
+		}
+
+		accumulateContainerValues()
+	}
+
+	if !IsCgroup2UnifiedMode() {
+		// cgroup v1 is in maintenance mode. Skip verifying pod cgroup
+		return nil
+	}
+
+	// Verify pod cgroup
+
+	verifyPodCgroupValue := func(cgPath, expectedCgValue string) error {
+		cgValueBytes, err := os.ReadFile(cgPath)
+		if err != nil {
+			return fmt.Errorf("failed to read pod cgroup value %q", cgPath)
+		}
+		cgValue := strings.Trim(string(cgValueBytes), "\n")
+		if cgValue != expectedCgValue {
+			return fmt.Errorf("pod cgroup value '%s' not equal to expected '%s'", cgValue, expectedCgValue)
+		}
+		return nil
+
+	}
+
+	expectedPodCPUShares := int64(kubecm.MilliCPUToShares(podCPURequestMilliValue))
+	expectedPodCPUQuota := int64(-1)
+	if podCPULimitMilliValue > 0 {
+		expectedPodCPUQuota = kubecm.MilliCPUToQuota(podCPULimitMilliValue, kubecm.QuotaPeriod)
+	}
+	expectedPodCPULimitString := strconv.FormatInt(expectedPodCPUQuota, 10)
+	if podMemoryLimitInBytes < 0 {
+		podMemoryLimitInBytes = 0
+	}
+	expectedPodMemLimitString := strconv.FormatInt(podMemoryLimitInBytes, 10)
+	if podOnCgroupv2Node {
+		if expectedPodCPULimitString == "-1" {
+			expectedPodCPULimitString = "max"
+		}
+		expectedPodCPULimitString = fmt.Sprintf("%s %s", expectedPodCPULimitString, CPUPeriod)
+		if expectedPodMemLimitString == "0" {
+			expectedPodMemLimitString = "max"
+		}
+		// convert cgroup v1 cpu.shares value to cgroup v2 cpu.weight value
+		// https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/2254-cgroup-v2#phase-1-convert-from-cgroups-v1-settings-to-v2
+		expectedPodCPUShares = 1 + ((expectedPodCPUShares-2)*9999)/262142
 	}
+
+	var cgroupName kubecm.CgroupName
+	switch pod.Status.QOSClass {
+	case v1.PodQOSGuaranteed:
+		cgroupName = kubecm.NewCgroupName(kubecm.RootCgroupName, defaultNodeAllocatableCgroup, "pod"+string(pod.UID))
+	case v1.PodQOSBurstable:
+		cgroupName = kubecm.NewCgroupName(kubecm.RootCgroupName, defaultNodeAllocatableCgroup, "burstable", "pod"+string(pod.UID))
+	case v1.PodQOSBestEffort:
+		cgroupName = kubecm.NewCgroupName(kubecm.RootCgroupName, defaultNodeAllocatableCgroup, "besteffort", "pod"+string(pod.UID))
+	}
+	cgroupFsName := ""
+	if kubeletCfg.CgroupDriver == "systemd" {
+		cgroupFsName = cgroupName.ToSystemd()
+	} else {
+		cgroupFsName = cgroupName.ToCgroupfs()
+	}
+
+	subsystems, err := kubecm.GetCgroupSubsystems()
+	if err != nil {
+		return err
+	}
+
+	if err := verifyPodCgroupValue(filepath.Join(subsystems.MountPoints["cpu"], cgroupFsName, "cpu.weight"), strconv.FormatInt(expectedPodCPUShares, 10)); err != nil {
+		return err
+	}
+	if err := verifyPodCgroupValue(filepath.Join(subsystems.MountPoints["cpu"], cgroupFsName, "cpu.max"), expectedPodCPULimitString); err != nil {
+		return err
+	}
+	if err := verifyPodCgroupValue(filepath.Join(subsystems.MountPoints["memory"], cgroupFsName, "memory.max"), expectedPodMemLimitString); err != nil {
```

Comment:
I'm leaving it unfixed now because I hesitate to add `CgroupRoot = ""` to `pkg/kubelet/cm/util/cgroups_unsupprted.go` to pass typecheck only for tests.